desugarer.go

```diff
@@ -199,6 +199,17 @@ func desugarFields(location LocationRange, fields *astObjectFields, objLevel int
 	return nil
 }
 
+func desugarArrayComp(astComp *astArrayComp, objLevel int) (err error) {
```

Comment:
If you like, I think you can use a bool for the objLevel (i.e. abstract it to == 0 or > 0).